egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,470 @@
+#!/usr/bin/env python
+# Copyright 2016  Tom Ko
+# Apache 2.0
+# script to generate reverberated data
+
+# we're using python 3.x style print but want it to work in python 2.x,
+from __future__ import print_function
+import argparse, glob, math, os, random, sys, warnings, copy, imp, ast
+
+data_lib = imp.load_source('ntl', 'steps/data/data_lib.py')
+
+def GetArgs():
+    # we add compulsary arguments as named arguments for readability
+    parser = argparse.ArgumentParser(description="Reverberate the data directory with an option "
+                                                 "to add isotropic and point source noiseis. "
+                                                 "Usage: reverberate_data_dir.py [options...] <in-data-dir> <out-data-dir> "
+                                                 "E.g. reverberate_data_dir.py --rir-list-file rir_list "
+                                                 "--foreground-snrs 20:10:15:5:0 --background-snrs 20:10:15:5:0 "
+                                                 "--noise-list-file noise_list --speech-rvb-probability 1 --num-replications 2 "
+                                                 "--random-seed 1 data/train data/train_rvb",
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument("--rir-list-file", type=str, required = True, 
+                        help="RIR information file, the format of the file is "
+                        "--rir-id <string,compulsary> --room-id <string,compulsary> "
+                        "--receiver-position-id <string,optional> --source-position-id <string,optional> "
+                        "--rt-60 < <float,optional> --drr <float, optional> < location(support Kaldi IO strings) >")
+    parser.add_argument("--noise-list-file", type=str, default = None,
+                        help="Noise information file, the format of the file is"
+                        "--noise-id <string,compulsary> --noise-type <choices = (isotropic, point source),compulsary> "
```

Comment:
btw, it's spelled "compulsory", but I would use the word "required" instead.
Dan


On Tue, Jul 19, 2016 at 8:14 PM, tomkocse <notifications@github.com> wrote:

> In egs/wsj/s5/steps/data/reverberate_data_dir.py
> <https://github.com/kaldi-asr/kaldi/pull/706#discussion_r71459791>:
>
> > +                                                 "to add isotropic and point source noiseis. "
> > +                                                 "Usage: reverberate_data_dir.py [options...] <in-data-dir> <out-data-dir> "
> > +                                                 "E.g. reverberate_data_dir.py --rir-list-file rir_list "
> > +                                                 "--foreground-snrs 20:10:15:5:0 --background-snrs 20:10:15:5:0 "
> > +                                                 "--noise-list-file noise_list --speech-rvb-probability 1 --num-replications 2 "
> > +                                                 "--random-seed 1 data/train data/train_rvb",
> > +                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
> > +
> > +    parser.add_argument("--rir-list-file", type=str, required = True,
> > +                        help="RIR information file, the format of the file is "
> > +                        "--rir-id <string,compulsary> --room-id <string,compulsary> "
> > +                        "--receiver-position-id <string,optional> --source-position-id <string,optional> "
> > +                        "--rt-60 < <float,optional> --drr <float, optional> < location(support Kaldi IO strings) >")
> > +    parser.add_argument("--noise-list-file", type=str, default = None,
> > +                        help="Noise information file, the format of the file is"
> > +                        "--noise-id <string,compulsary> --noise-type <choices = (isotropic, point source),compulsary> "
>
> I saw that in your other script you use "[]" for choices , so which one
> you prefer?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/706/files/d335c718e3b1849afe40a61fc436c0e098c6a36e#r71459791>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu54zkMntG4E2J0t9USNTsuY-JQygks5qXZKwgaJpZM4IJM2C>
> .
>